include/coap3/coap_oscore.h

```diff
@@ -129,12 +129,17 @@ int coap_context_oscore_server(coap_context_t *context,
  *
  * @param sender_seq_num The Sender Sequence Number to save in non-volatile
  *                      memory.
+ * @param recipient_id Recipient ID that could be used to find the corresponding SSN entry.
+ * @param id_context ID Context that could be used to find the corresponding SSN entry.
  * @param param The save_seq_num_func_param provided to
  *              coap_new_oscore_context().
  *
  * @return @c 1 if success, else @c 0 if a failure of some sort.
  */
-typedef int (*coap_oscore_save_seq_num_t)(uint64_t sender_seq_num, void *param);
+typedef int (*coap_oscore_save_seq_num_t)(uint64_t sender_seq_num, 
+                                          const coap_bin_const_t *recipient_id, 
+                                          const coap_bin_const_t *id_context,
+                                          void *param);
```

Comment:
This is a change to the Public API which we do not want to have to do unless it is absolutely critical.

My suggestion would be to add in save (and delete) function parameters to `coap_register_oscore_context_handler()` and call those functions in the appropriate points in the code.  Then from a user perspective, everything is in one place for doing the definitions.

Update coap_context-t to include something similar to external_oscore_update_context_handler and external_oscore_delete_context_handler.